metomi/rose/macros/rule.py

```diff
@@ -50,6 +50,198 @@
 )
 
 
+class MyInt(int):
+    """Override integer to maintain Python2 style interface
+    """
+    def __lt__(self, other):
+        """
+        Examples:
+            >>> MyInt(4) < MyInt(6)
+            True
+            >>> MyInt(4) < MyFloat(6.1)
+            True
+            >>> MyInt(4) < MyStr('Zaphod Beeblebrox')
+            True
+            >>> MyInt(99999) < MyStr('Zaphod Beeblebrox')
+            True
+            >>> MyInt(4) < MyFloat(-5.5)
+            False
+            >>> MyInt(42) < 42
+            False
+            >>> MyInt(77) < ''
+            False
+        """
+        if isinstance(other, (int, float, MyFloat, MyInt)):
+            return float(self) < float(other)
+        elif isinstance(other, MyStr):
+            return True
+        else:
+            return int(self) == other
+
+    def __eq__(self, other):
+        """
+        Examples:
+            >>> MyInt(22) == MyInt(22)
+            True
+            >>> MyInt(22) == MyInt(44)
+            False
+            >>> MyInt(22) == MyFloat(4.75734)
+            False
+            >>> MyInt(22) == MyFloat(22.0)
+            True
+            >>> MyInt(123) == MyStr('Hello World')
+            False
+            >>> MyInt(-77) == 'Viltvodle VI'
+            False
+        """
+        if isinstance(other, (int, float, MyFloat, MyInt)):
+            return float(self) == float(other)
+        elif isinstance(other, MyStr):
+            return False
+        else:
+            return int(self) == other
```

Comment:
Another option here might be:

```python
try:
    return self == other
except TypeError:
    return False
```

I think this would be more robust and avoid casting to `float` which could alter results?

Ditto for the others?